packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.tsx

```diff
@@ -0,0 +1,580 @@
+          // is incountered (or if we reach the end of the items))
+          elements.push(this._renderNextChuckOfCellItems(
+            items.slice(index),
+            posInSet,
+            setSize));
+          break;
+        default:
+
+          // Does at least one of the menu items have an icon?
+          if (shouldAccountForIcon) {
+
+            // Make sure the width styling is the same by applying a consistent className
+            if (item.menuItemButtonProps) {
+              item.menuItemButtonProps = {
+                ...item.menuItemButtonProps,
+                iconProps: { ...item.menuItemButtonProps.iconProps, className: styles.icon }
+              };
+            } else {
+
+              // This menu item didn't have an icon so add a "spacer" icon to make the text
+              // content align. This aligns with how ContextualMenu achieves this alignmet
+              item.menuItemButtonProps = { ...item.menuItemButtonProps, iconProps: { iconName: 'customIcon', className: styles.icon } };
+            }
+          }
+          elements.push(this._renderOption(item, posInSet, setSize));
+      }
+
+      // Increase the index by the number of items in the just processed chunk,
+      // otherwise just increment the index
+      index += this._numOfItemsInChunk > 0 ? this._numOfItemsInChunk : 1;
+      this._numOfItemsInChunk = -1;
+    }
+
+    return elements;
+  }
+
+  /**
+   * Renders the next consecutive chunk of cells
+   * @param items - The items to process
+   * @param posInSet - The position in the set for this chunk
+   * @param setSize - The size of the total set
+   * @returns {JSX.Element} - The grid that represents the chunk
+   */
+  private _renderNextChuckOfCellItems(
+    items: ISwatchColorPickerItemProps[],
+    posInSet: number = null,
+    setSize: number = null): JSX.Element {
+    let chunkItems = this._getNextChuckOfCellItems(items);
+
+    // Update the number of items in chunk
+    this._numOfItemsInChunk = chunkItems.length > 0 ? chunkItems.length : -1;
+
+    return (
+      <Grid
+        key={ this._id + items[0].id + '-grid' }
+        items={ chunkItems }
+        width={ this.props.width }
+        onRenderItem={ this._renderOption }
+        positionInSet={ posInSet }
+        setSize={ setSize }
+      />
+    );
+  }
+
+  /**
+   * Get the position in set for the given id
+   * @param firstExecutableItemsPerChunk - The list of fist executable items per chunk
+   * @param itemId - The id of item to find the index of
+   * @returns {number} - The position in the set
+   */
+  private _getPositionInSet(firstExecutableItemsPerChunk: ISwatchColorPickerItemProps[], itemId: string): number {
+
+    // Find the index of the given id in the list of first executable items per chunk
+    let index = findIndex(firstExecutableItemsPerChunk, (executableItem) => (executableItem.id === itemId));
+
+    return index < 0 ? null : index + 1;
```

Comment:
This would not fly if fabric were doing strict null checks, which we want to do eventually. You should be explicit with the return type, number | null, or even better not return null.